torch/csrc/jit/codegen/cuda/scheduler/registry.cpp

```diff
@@ -713,87 +734,155 @@ class NormalizationScheduler : public SchedulerEntry {
+          if (!id->isBroadcast()) {
+            count++;
+          }
+        }
+        return count;
+      };
+
+      for (auto red : reduction_tvs) {
+        if (!valid_axis_count) {
+          valid_axis_count = true;
+          axis_count = reduction_root_size(red);
+        } else {
+          if (reduction_root_size(red) != axis_count) {
+            return false;
+          }
         }
       }
-      return count;
-    };
 
-    for (auto red : reduction_tvs) {
-      if (!valid_axis_count) {
-        valid_axis_count = true;
-        axis_count = reduction_root_size(red);
-      } else {
-        if (reduction_root_size(red) != axis_count) {
+      // Use root domain map to check the reduction ops have the same axes
+      FusionGuard fg(fusion);
+      ComputeAtRootDomainMap root_map;
+      root_map.build(true);
+
+      // red_ops.size()>1 checked before
+      for (size_t it = 1; it < reduction_tvs.size(); it++) {
+        if (!checkEquivalence(
+                reduction_tvs[it - 1], reduction_tvs[it], root_map)) {
           return false;
         }
       }
     }
 
-    // Use root domain map to check the reduction ops have the same axes
-    FusionGuard fg(fusion);
-    ComputeAtRootDomainMap root_map;
-    root_map.build(true);
-
-    // red_ops.size()>1 checked before
-    for (size_t it = 1; it < reduction_tvs.size(); it++) {
-      if (!checkEquivalence(
-              reduction_tvs[it - 1], reduction_tvs[it], root_map)) {
-        return false;
+    // TODO: move all these boilerplate code into the accessor class
+    // (follow up)
+    // Note: this persistent buffer is actually cached from
+    //  getNormalizationHeuristics. Will need to create a separate
+    //  cache entry if they are not the same.
+    HeuristicCacheAccessor<scheduler_utils::PersistentBufferInfo>
+        persistent_buffer_data;
+
+    if (data_cache && !data_cache->isRecording()) {
+      persistent_buffer_data.writeTemporary(
+          data_cache->getPersistentBufferInfo());
+    } else {
+      persistent_buffer_data.writeNew(
+          scheduler_utils::persistentBuffers(fusion));
+      if (data_cache && data_cache->isRecording()) {
+        data_cache->setPersistentBufferInfo(persistent_buffer_data.read());
       }
     }
+    auto& persistent_buffers = persistent_buffer_data.read();
 
-    auto persistent_buffer_size =
-        scheduler_utils::persistentBufferSize(fusion, runtime_info);
+    auto persistent_buffer_size = scheduler_utils::persistentBufferSize(
+        fusion, runtime_info, persistent_buffers, data_cache);
     if (persistent_buffer_size * 4 > scheduler_utils::register_file_size * 3) {
       return false;
     }
 
+    // TODO: really need to make inserting an entry into data_cache easier to do
+    HeuristicCacheAccessor<bool> has_post_reduction_bcast_data;
+
+    if (data_cache && !data_cache->isRecording()) {
+      has_post_reduction_bcast_data.writeTemporary(
+          data_cache->getHasPostReductionBCast());
```

Comment:
Is this also in the single reduction can schedule?

Reply:
Yes single reduction canSchedule checks this as well but not in a way that's runtime info dependent, which I assume means we check the same graph again at the runtime we get the same result. 

The reason we I cached these for normalization schedulers is because of the logic in line 880, which is runtime dependent use of the info, and so we need to have these boolean values ready at runtime.  Same for `getSupportedPostReductionFusion`. I need to look closer to vectorizable inputs and outputs.